mllib/src/main/scala/org/apache/spark/ml/feature/MinHashLSH.scala

```diff
@@ -109,9 +109,16 @@ class MinHashLSHModel private[ml](
   override protected[ml] def hashDistance(x: Seq[Vector], y: Seq[Vector]): Double = {
     // Since it's generated by hashing, it will be a pair of dense vectors.
     // TODO: This hashDistance function requires more discussion in SPARK-18454
-    x.iterator.zip(y.iterator).map(vectorPair =>
-      vectorPair._1.toArray.zip(vectorPair._2.toArray).count(pair => pair._1 != pair._2)
-    ).min
+    // Currently each hash vector (generated by hashFunction) only has one element, this equals to:
+    // x.iterator.zip(y.iterator).map(vectorPair =>
+    //   vectorPair._1.toArray.zip(vectorPair._2.toArray).count(pair => pair._1 != pair._2)
+    // ).min
+    val iter = x.iterator.zip(y.iterator)
+    while (iter.hasNext) {
+      val (vx, vy) = iter.next
+      if (vx(0) == vy(0)) return 0.0
+    }
+    1.0
```

Comment:
Remind me why this must be the min if no elements match?

Reply:
Since each vector has only one value, so in 

```
    x.iterator.zip(y.iterator).map(vectorPair =>
      vectorPair._1.toArray.zip(vectorPair._2.toArray).count(pair => pair._1 != pair._2)
    ).min
```

the `vectorPair._1.toArray.zip(vectorPair._2.toArray)` has only one (Double, Double) tuple, the `.count(pair => pair._1 != pair._2)` must be 0 or 1.